docs/grammar/SolidityParser.g4

```diff
@@ -409,9 +412,12 @@ inlineArrayExpression: LBrack (expression ( Comma expression)* ) RBrack;
 /**
  * Besides regular non-keyword Identifiers, some keywords like 'from' and 'error' can also be used as identifiers.
  */
-identifier: Identifier | From | Error | Revert | Global;
+identifier: Identifier | From | Error | Revert | Global | Suffix;
+
+suffixedLiteral: (stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral) identifier;
+
+literal: stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral | numberLiteralWithSubdenomination;
```

Comment:
We now have a bit of duplication here. Maybe something like this would be better?

```suggestion
literal: stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral;

suffixedLiteral: literal identifier;
literalWithSubdenomination: numberLiteral Subdenomination;
```

Then we'd explicitly list all variants of a literal in `expression` below:
```
		| literal
		| literalWithSubdenomination
		| suffixedLiteral
```

This might also be more readable because, personally, I'd not expect something like `1 gwei` to be parsed as a literal. I'd expect a separate rule for it.

Reply:
Yes, that's much better and easier to read. Done.